pyuvdata/uvflag.py

```diff
@@ -0,0 +1,832 @@
+
+    def to_antenna(self, uv, force_pol=False):
+        '''Convert a UVFlag object of type "waterfall" to type "antenna".
+        Broadcasts the flag array to all antennas.
+        This function does NOT apply flags to uv.
+        Args:
+            uv: UVCal or UVFlag object of type antenna to match.
+            force_pol: If True, will use 1 pol to broadcast to any other pol.
+                       Otherwise, will require polarizations match.
+        '''
+        if self.type == 'antenna':
+            return
+        if not (issubclass(uv.__class__, UVCal) or (isinstance(uv, UVFlag) and uv.type == 'antenna')):
+            raise ValueError('Must pass in UVCal object or UVFlag object of type '
+                             '"antenna" to match.')
+        if self.type != 'waterfall':
+            raise ValueError('Cannot convert from type "' + self.type + '" to "antenna".')
+        # Deal with polarization
+        if issubclass(uv.__class__, UVCal):
+            polarr = uv.jones_array
+        else:
+            polarr = uv.polarization_array
+        if force_pol and self.polarization_array.size == 1:
+            # Use single pol for all pols, regardless
+            self.polarization_array = polarr
+            # Broadcast arrays
+            if self.mode == 'flag':
+                self.flag_array = self.flag_array.repeat(self.polarization_array.size, axis=-1)
+            else:
+                self.metric_array = self.metric_array.repeat(self.polarization_array.size, axis=-1)
+            self.weights_array = self.weights_array.repeat(self.polarization_array.size, axis=-1)
+        # Now the pol axes should match regardless of force_pol.
+        if not np.array_equal(polarr, self.polarization_array):
+            raise ValueError('Polarizations could not be made to match.')
+        # Populate arrays
+        if self.mode == 'flag':
+            self.flag_array = np.swapaxes(self.flag_array, 0, 1)[np.newaxis, np.newaxis,
+                                                                 :, :, :]
+            self.flag_array = self.flag_array.repeat(len(uv.ant_array), axis=0)
+        elif self.mode == 'metric':
+            self.metric_array = np.swapaxes(self.metric_array, 0, 1)[np.newaxis, np.newaxis,
+                                                                     :, :, :]
+            self.metric_array = self.metric_array.repeat(len(uv.ant_array), axis=0)
+        self.weights_array = np.swapaxes(self.weights_array, 0, 1)[np.newaxis, np.newaxis,
+                                                                   :, :, :]
+        self.weights_array = self.weights_array.repeat(len(uv.ant_array), axis=0)
+        self.ant_array = uv.ant_array
+        self.type = 'antenna'
+        self.history += 'Broadcast to type "antenna" with ' + self.pyuvdata_version_str
+
+    def to_flag(self):
+        '''Convert to flag mode. NOT SMART. Simply removes metric_array and initializes
+        flag_array with Falses.
```

Comment:
Is there a method to threshold a metric to turn it into a flag array?

Reply:
Currently, this is done in `xrfi`, and the function here is just for converting the UVFlag mode. We could implement a very simple version here (a `threshold` keyword?), but I don't think we want to get into the business of flagging algorithms. What do you think?

Reply:
yeah, this probably belongs in analysis code. If someone wants it in the future it can be added then.

Reply:
A very simple "threshold" version would also be totally reasonable.